heath/h19.cpp Implement Shift-Reset functionality

mgarlanger opened the pull request:
The H19 allowed the user to press the right-shift and reset key to reset the system. Since the h89 shared the common terminal board in the H19, this reset also was routed to the h89 cpu board and reset the entire computer. The actual reset occurs on the release of the keys. 

In mame, the default key for reset is F10. Using the right-shift and F10 now causes the machine (either H19 or H89) to be reset.

A few other general cleanup of comments, formatting, etc. is also done in this PR.

cuavas commented on src/mame/heathkit/tlb.cpp:
```diff
@@ -153,26 +157,40 @@ static constexpr uint8_t KB_STATUS_KEYBOARD_STROBE_MASK = 0x80;
+	m_bellactive = false;
+	m_reset_pending = false;
+
+	m_crtc->reset();
+	m_ace->reset();
+	m_beep->reset();
+	m_mm5740->reset();
+	m_maincpu->reset();
```
Once again, you shouldn’t be resetting child devices from `device_reset`.

cuavas commented on src/mame/heathkit/tlb.cpp:
```diff
@@ -242,16 +277,18 @@ uint8_t heath_tlb_device::kbd_flags_r()
+	return (m_kbspecial->read() & 0x10) ? CLEAR_LINE : ASSERT_LINE;
 }
 
 void heath_tlb_device::mm5740_data_ready_w(int state)
 {
+	check_for_reset();
+
```
Is this when the reset key combination is actually checked, or does it happen out-of-band?  If it happens out-of-band, you should attach callbacks to the relevant keys so you get a notification when they’re changed.

cuavas commented on src/mame/heathkit/intr_cntrl.cpp:
```diff
@@ -31,6 +31,11 @@ void heath_intr_cntrl::device_start()
 	m_intr_lines = 0;
 }
 
+void heath_intr_cntrl::device_reset()
+{
+	m_intr_lines = 0;
+}
+
```
This value comes from inputs.  You shouldn’t change it on reset.  If the devices that produce those inputs deassert their outputs on reset, it’s their responsibility to update the outputs on reset.

cuavas commented on src/mame/heathkit/h89.cpp:
```diff
@@ -362,9 +363,15 @@ void h89_state::machine_reset()
 
 	update_gpp(0);
 	update_mem_view();
+	m_h37->reset();
+	m_intr_cntrl->reset();
+	m_console->reset();
+	m_serial1->reset();
+	m_serial2->reset();
+	m_serial3->reset();
+	m_maincpu->reset();
```
Once again, you shouldn’t be resetting child devices here – that happens automatically.

cuavas commented on src/mame/heathkit/h89.cpp:
```diff
@@ -389,6 +396,14 @@ void h89_state::console_intr(uint8_t data)
 	}
 }
 
+void h89_state::reset_line(uint8_t data)
+{
+	if (data == ASSERT_LINE)
+	{
+		reset();
+	}
+}
```
Read/write line members conventionally use `int` values.

cuavas commented on src/mame/heathkit/intr_cntrl.cpp:
```diff
@@ -171,6 +176,15 @@ void z37_intr_cntrl::device_start()
+{
+	heath_intr_cntrl::device_reset();
+
+	m_interrupts_blocked = false;
+	m_drq_raised = false;
+	m_fd_irq_raised = false;
```
You shouldn’t be changing inputs on reset.  If the devices connected to those inputs deassert them on reset, it’s their responsibility to push the change.

cuavas reviewed (commented):
There’s too much wrong with this.  I’m going to revert it.

happppp commented:
FYI to do a full system reset:
machine().schedule_soft_reset();

cuavas commented:
> FYI to do a full system reset: machine().schedule_soft_reset();

But that will also reset things attached to serial ports, etc.  Calling `reset()` on the root device has almost the same effect, although it doesn’t wait to return to the scheduler, which can upset some CPUs.

mgarlanger commented:
Reverting seems a little excessive when the machine runs fine with the new functionality _and_ there is existing code that does the same things that this PR does wrong in such popular systems as C64/C128- https://github.com/mamedev/mame/blob/master/src/mame/commodore/c64.cpp#L1447 (and there are may other examples of it in the code). 

Anyways, new PR https://github.com/mamedev/mame/pull/11339 addresses the comments raised in this PR.
